osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -30,12 +30,21 @@ public ReplayRecorder Recorder
         {
             set
             {
-                if (recorder != null)
-                    throw new InvalidOperationException("Cannot attach more than one recorder");
-
-                recorder = value;
-
-                KeyBindingContainer.Add(recorder);
+                if (value == null)
+                {
+                    if (recorder == null) return;
+
+                    KeyBindingContainer.Remove(recorder);
+                    recorder.Dispose();
+                }
+                else
+                {
+                    if (recorder != null)
+                        throw new InvalidOperationException("Cannot attach more than one recorder");
+
+                    recorder = value;
+                    KeyBindingContainer.Add(recorder);
+                }
```

Comment:
I'm not a 100% fan of this "set null to stop recording" logic, but having this as a property that can be initialised with the rest of the hierarchy works nice for test code, so maybe fine to have this...

Reply:
Yeah, not 100% on it. Although you could think of it less as "to stop recording" and more just "disposing the old instance".. not sure.